migration/handler_test.go

```diff
@@ -0,0 +1,120 @@
+package migration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestSchemaMigratingHandler(t *testing.T) {
+	const thisPkgName = "testpkg"
+
+	reg := newRegister()
+
+	reg.MustRegister(1, &MyMsg{}, NoModification)
+	reg.MustRegister(2, &MyMsg{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
```

Comment:
Ahh... so you edit the message in place.
Hmmm... Somehow I prefer passing a new one, but the truth is this makes it much cleaner to reuse the same transaction.

Reply:
I is not possible to implement it with a copy making. There is `Copy()` method, but because of types working with such code would be very hard. It would be also hard to update the references to message/model from old to new one (ie for transaction).